src/test/java/hudson/plugins/git/IndexEntryTest.java

```diff
@@ -137,11 +135,6 @@ public void testEquals() {
         assertEquals(entryNull4, entryNull4a);
         assertEquals(entryNull4a, entryNull4);
 
-        IndexEntry entryNullA = null;
-        IndexEntry entryNullB = null;
-        assertEquals(entryNullA, entryNullB);
```

Comment:
I think that is a potentially useful test of the equals contract, isn't it?  

Two objects of type `IndexEntry` which have the same value (`null`) are tested for equality.

Reply:
The class's `equals` method is not called here. That's because inside JUnit's assertion methods a shortcut for `actual` and `expected` beeing both `null` is implemented. The same holds true for Hamcrest matchers - e.g. `assertThat(entryNullA, is(equalTo(entryNullB)));`.